include/os/linux/spl/sys/list.h

```diff
@@ -50,6 +50,7 @@ typedef struct list_head list_node_t;
 typedef struct list {
 	size_t list_size;
 	size_t list_offset;
+	uint32_t list_length;
```

Comment:
Considering all types around and similar to AVL-trees, ulong_t would make more sense.